src/main.js

```diff
@@ -18,154 +20,307 @@ import {
+  const infoPostComment = formatPostComment({
+    input: info,
+    type: "INFO",
+  });
+  await commentServices.postComment(
+    octokit,
+    prData.baseOwner,
+    prData.baseRepo,
+    prData.prNumber,
+    infoPostComment
+  );
+
+  await handleLabels(octokit, prData, "remove-all-labels");
+};
+
+const postErrorMessageAboutMissingChangesetFile = async (
+  error,
+  octokit,
+  prData
+) => {
+  console.log(
+    `Changeset file ${prData.prNumber}.yml is missing in the forked repository.`
+  );
 
-    // Delete changeset file if one was previously created
-    if (
-      error.name !== "GitHubAppSuspendedOrNotInstalledError" &&
-      error.name !== "MissingChangelogPullRequestBridgeUrlDomainError" &&
-      error.name !== "MissingChangelogPullRequestBridgeApiKeyError" &&
-      error.name !== "UnauthorizedRequestToPullRequestBridgeServiceError"
-    ) {
-      const commitMessage = `Changeset file for PR #${prNumber} deleted`;
-      await forkedFileServices.deleteFileInForkedRepoByPath(
-        headOwner,
-        headRepo,
-        headBranch,
-        changesetFilePath(prNumber),
-        commitMessage
+  const errorPostComment = formatPostComment({
+    input: error,
+    type: "ERROR",
+  });
+  await commentServices.postComment(
+    octokit,
+    prData.baseOwner,
+    prData.baseRepo,
+    prData.prNumber,
+    errorPostComment
+  );
+  await handleLabels(octokit, prData, "add-failed-label");
+};
+
+// ----------------------------------------------------------
+// Labels Helpers Functions
+// ----------------------------------------------------------
+const handleLabels = async (octokit, prData, operation) => {
+  switch (operation) {
+    case "add-skip-label":
+      await labelServices.addLabel(
+        octokit,
+        prData.baseOwner,
+        prData.baseRepo,
+        prData.prNumber,
+        SKIP_LABEL
       );
-    }
-    throw new Error("Changeset creation workflow failed.");
+      await labelServices.removeLabel(
+        octokit,
+        prData.baseOwner,
+        prData.baseRepo,
+        prData.prNumber,
+        FAILED_CHANGESET_LABEL
+      );
+      break;
+    case "add-failed-label":
+      await labelServices.addLabel(
+        octokit,
+        prData.baseOwner,
+        prData.baseRepo,
+        prData.prNumber,
+        FAILED_CHANGESET_LABEL
+      );
+      await labelServices.removeLabel(
+        octokit,
+        prData.baseOwner,
+        prData.baseRepo,
+        prData.prNumber,
+        SKIP_LABEL
+      );
+      break;
+    case "remove-all-labels":
+      await labelServices.removeLabel(
+        octokit,
+        prData.baseOwner,
+        prData.baseRepo,
+        prData.prNumber,
+        SKIP_LABEL
+      );
+      await labelServices.removeLabel(
+        octokit,
+        prData.baseOwner,
+        prData.baseRepo,
+        prData.prNumber,
+        FAILED_CHANGESET_LABEL
+      );
+      break;
+    default:
+      console.log(`Unknown operation: ${operation}`);
```

Comment:
nit: seems like an error to me
```suggestion
      console.error(`Unknown operation: ${operation}`);
```

Reply:
Solved